airflow-core.yaml

```diff
@@ -0,0 +1,149 @@
+      - py3.12-pip
+      - python-3.12
+      - python-3.12-dev
+      - wolfi-base
+      - zlib-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/airflow
+      tag: ${{package.version}}
+      expected-commit: c083e456fa02c6cb32cdbe0c9ed3c3b2380beccd
+
+  - runs: |
+      # Set required build environment variables
+      export SOURCE_DATE_EPOCH=315532800
+      export MYSQLCLIENT_CFLAGS=$(mysql_config --cflags)
+
+      # Install airflow core, the upstream sets this as directory
+      export PYTHONUSERBASE=${{targets.destdir}}/home/airflow/.local
+      pip install --no-cache-dir --user apache-airflow[core]==${{package.version}}
+
+      #GHSA-8w49-h785-mj3c/GHSA-8495-4g3g-x7pr/GHSA-27mf-ghqm-j3j8 fixes
+      pip install --no-cache-dir --user aiohttp>=3.10.11 tornado>=6.4.2
+      pip install --no-cache-dir --user --force-reinstall packaging statsd
+
+  - runs: |
+      # CVE-2024-6345 GHSA-cx63-2mw6-8hw5
+      # setuptools comes from airflow/providers/google/provider.yaml having
+      # gcloud-aio-auth>=4.0.0,<5.0.0 . gcloud-aio-auth 4 is backlevel and has
+      # setuptools in it's pyproject.toml 'tool.poetry.dependencies'
+      # The tldr; For that case it is not needed in runtime.
+      pip uninstall --yes setuptools
+      find . -name '__pycache__' -exec rm -rf {} +
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/opt/airflow/dags
+      mkdir -p ${{targets.destdir}}/opt/airflow/logs
+      mkdir -p ${{targets.destdir}}/scripts/docker
+
+      # The first time you run Airflow, it will create a file called `airflow.cfg` in
+      # `$AIRFLOW_HOME` directory
+      # However, for production case it is advised to generate the configuration
+      export PYTHONPATH=${{targets.destdir}}/home/airflow/.local/lib/python3.12/site-packages
+
+      ${{targets.destdir}}/home/airflow/.local/bin/airflow config list --defaults > ${{targets.destdir}}/opt/airflow/"airflow.cfg"
+
+      cp airflow/config_templates/default_webserver_config.py ${{targets.destdir}}/
+
+      cp scripts/docker/entrypoint_prod.sh ${{targets.destdir}}/entrypoint
+      chmod 755 ${{targets.destdir}}/entrypoint
+
+      cp scripts/docker/clean-logs.sh ${{targets.destdir}}/clean-logs
+      chmod 755 ${{targets.destdir}}/clean-logs
+
+      cp scripts/docker/airflow-scheduler-autorestart.sh ${{targets.destdir}}/airflow-scheduler-autorestart
+      chmod 755 ${{targets.destdir}}/airflow-scheduler-autorestart
+
+      # unlike airflow, airflow-core needs only these scripts
+      cp scripts/docker/common.sh ${{targets.destdir}}/scripts/docker
+      cp scripts/docker/install_mssql.sh ${{targets.destdir}}/scripts/docker
+      cp scripts/docker/install_mysql.sh ${{targets.destdir}}/scripts/docker
+      cp scripts/docker/install_os_dependencies.sh ${{targets.destdir}}/scripts/docker
+      cp scripts/docker/install_postgres.sh ${{targets.destdir}}/scripts/docker
+      chmod 755 ${{targets.destdir}}/scripts/docker/*
```

Comment:
In the past, we've typically split this into a separate entrypoint package, under the assumption that the package itself can operate independently

Reply:
Done, splitted this in another package